src/Zed/Persistence/Repository/RepositoryPropelQueryJoinRule.php

```diff
@@ -0,0 +1,254 @@
+     */
+    protected function getFacade(): PropelQueryFacadeInterface
+    {
+        if (!$this->facade) {
+            $this->facade = new PropelQueryFacade();
+        }
+
+        return $this->facade;
+    }
+
+    /**
+     * @param string $methodName
+     * @param string $relationTableName
+     * @param string $moduleName
+     * @param \PHPMD\Node\ClassNode $node
+     *
+     * @return void
+     */
+    protected function addViolationMessage(string $methodName, string $relationTableName, string $moduleName, ClassNode $node): void
+    {
+        $message = sprintf(
+            'The Repository method %s() violates rule "%s". ',
+            $methodName,
+            static::RULE
+        );
+
+        $solutionMessage = sprintf(
+            'Please add @module %s for %s join.',
+            $moduleName,
+            $relationTableName
+        );
+
+        $message .= $solutionMessage;
+
+        $this->addViolation($node, [$message]);
+    }
+
+    /**
+     * @param string $relationTableName
+     * @param \ArchitectureSniffer\PropelQuery\Schema\Transfer\PropelSchemaTableTransfer[] $tableTransfers
+     *
+     * @return \ArchitectureSniffer\PropelQuery\Schema\Transfer\PropelSchemaTableTransfer $tableTransfer
+     */
+    protected function findRelationTableTransfer(string $relationTableName, array $tableTransfers): ?PropelSchemaTableTransfer
+    {
+        if ($tableTransfers === []) {
+            return null;
+        }
+
+        foreach ($tableTransfers as $tableTransfer) {
+            $phpName = $tableTransfer->getPhpName();
+
+            if ($relationTableName === $phpName) {
+                return $tableTransfer;
+            }
+
+            $relations = $tableTransfer->getRelations();
+
+            foreach ($relations as $relation) {
+                $relationPhpName = $relation->getPhpName();
+                $referencePhpName = $relation->getRefPhpName();
+
+                if ($relationPhpName === $relationTableName) {
+                    return $tableTransfers[$relation->getTableName()];
+                }
+
+                if ($referencePhpName === $relationTableName) {
+                    return $tableTransfer;
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * @param \PHPMD\Node\ClassNode $node
+     *
+     * @return \ArchitectureSniffer\PropelQuery\ClassNode\Transfer\ClassNodeTransfer
+     */
+    protected function getClassNodeTransfer(ClassNode $node): ClassNodeTransfer
+    {
+        $className = $node->getName();
+        $classNamespace = $node->getNamespaceName();
+        $classFilePath = $node->getFileName();
+        $classModuleName = $this->getFacade()->getModuleName($node);
+
+        $pathTransfer = $this->getFacade()->getPath($classFilePath);
+        $reflectionNodeClass = $this->getFacade()->getReflectionClassByNode($node);
+
+        $classNodeTransfer = new ClassNodeTransfer();
+
+        $classNodeTransfer->setClassName($className);
```

Comment:
Please make use of fluent interface here and for the other transfers added by you.